pkg/inventory/sysfs.go

```diff
@@ -40,6 +41,10 @@ const (
 	sysdevPath = "/sys/devices"
 )
 
+// pciAddressRegex is used to identify a PCI address within a string.
+// It matches patterns like "0000:00:04.0" or "00:04.0".
+var pciAddressRegex = regexp.MustCompile(`^(?:([0-9a-fA-F]{4}):)?([0-9a-fA-F]{2}):([0-9a-fA-F]{2})\.([0-9a-fA-F])$`)
```

Comment:
ok, this always returns 5 elements in a match package main
```
import (
	"fmt"
	"regexp"
)

func main() {
	re := regexp.MustCompile(`^(?:([0-9a-fA-F]{4}):)?([0-9a-fA-F]{2}):([0-9a-fA-F]{2})\.([0-9a-fA-F])$`)
	fmt.Printf("%q\n", re.FindStringSubmatch("0000:00:04.0"))
	fmt.Printf("%q\n", re.FindStringSubmatch("00:04.0"))
}
```
```
["0000:00:04.0" "0000" "00" "04" "0"]
["00:04.0" "" "00" "04" "0"]
```

https://go.dev/play/p/5pTJYFVKQKx